logger_block.py

```diff
@@ -33,11 +31,21 @@ def process_signals(self, signals):
             None
         """
         log_func = self._get_logger()
-        for s in signals:
+
+        # make it obvious that this was passed a list of signals or just one
+        # by logging a list or not.
+        if len(signals) > 1:
             try:
-                log_func(s)
+                log_func(signals)
```

Comment:
I wouldn't be opposed to just logging every list like this, even if it's just one signal.